src/context/ListContext.tsx

```diff
@@ -0,0 +1,201 @@
+  selectedUser: User | null;
+  isUsersLoading: boolean;
+  onSelectUser: (user: User) => void;
+}
+
+interface ICommentsContext {
+  comments: Comment[];
+  isCommentsLoading: boolean;
+  commentIsProcessing: boolean;
+  onAddNewComment: (comment: CommentData) => Promise<boolean>;
+  onDeleteComment: (commentId: number) => void;
+}
+
+export const GlobalContext = createContext({} as IGlobalContext);
+export const PostsContext = createContext({} as IPostsContext);
+export const UsersContext = createContext({} as IUsersContext);
+export const CommentsContext = createContext({} as ICommentsContext);
+
+export const ListProvider: FC<Props> = ({ children }) => {
+  const [error, setError] = useState(false);
+
+  // #region User states
+  const [users, setUsers] = useState<User[]>([]);
+  const [selectedUser, setSelectedUser] = useState<User | null>(null);
+  const [isUsersLoading, setIsUsersLoading] = useState(false);
+  // #endregion
+
+  // #region Posts states
+  const [posts, setPosts] = useState<Post[]>([]);
+  const [isPostsLoading, setIsPostsLoading] = useState(false);
+  const [selectedPost, setSelectedPost] = useState<Post | null>(null);
+  // #endregion
+
+  // #region Comments states
+  const [comments, setComments] = useState<Comment[]>([]);
+  const [isCommentsLoading, setIsCommentsLoading] = useState(false);
+  const [commentIsProcessing, setCommentIsProcessing] = useState(false);
+  // #endregion
+
+  // #region load data
+
+  /* load users */
+  useEffect(() => {
+    setError(false);
+    setIsUsersLoading(true);
+
+    getUsers()
+      .then((usersList) => setUsers(usersList))
+      .catch(() => setError(true))
+      .finally(() => setIsUsersLoading(false));
+  }, []);
+
+  /* load posts for selected user */
+  useEffect(() => {
+    setError(false);
+    setSelectedPost(null);
+
+    if (selectedUser) {
+      getPosts(selectedUser.id)
+        .then(postList => setPosts(postList))
+        .catch(() => setError(true))
+        .finally(() => setIsPostsLoading(false));
+    }
+  }, [selectedUser]);
+
+  /* load comments for selected post */
+  useEffect(() => {
+    setError(false);
+
+    if (selectedPost) {
+      getComments(selectedPost.id)
+        .then(commentsList => setComments(commentsList))
+        .catch(() => setError(true))
+        .finally(() => setIsCommentsLoading(false));
+    }
+  }, [selectedPost]);
+  // #endregion
+
+  const onSelectUser = (user: User) => {
+    if (selectedUser === user) {
+      return;
+    }
+
+    setSelectedUser(user);
+    setPosts([]);
+    setIsPostsLoading(true);
+  };
+
+  const onSelectPost = (post: Post) => {
+    setComments([]);
+
+    if (post.id === selectedPost?.id) {
+      setSelectedPost(null);
+
+      return;
+    }
+
+    setIsCommentsLoading(true);
+    setSelectedPost(post);
+  };
+
+  const onAddNewComment = (comment: CommentData) => {
+    setError(false);
+
+    setCommentIsProcessing(true);
+
+    return postComment((selectedPost?.id as number), comment)
```

Comment:
You should check if selectedPost?.id is not null before posting a new comment. If it is null, this code may crash.